src/Stripe.net/Infrastructure/Middleware/RequestStringBuilder.cs

```diff
@@ -1,61 +1,317 @@
 ﻿namespace Stripe.Infrastructure.Middleware
 {
+    using System;
+    using System.Collections;
     using System.Collections.Generic;
     using System.Globalization;
+    using System.Linq;
     using System.Net;
     using System.Reflection;
     using Newtonsoft.Json;
 
     public static class RequestStringBuilder
     {
-        static RequestStringBuilder()
+        public static void ApplyParameterToRequestString(ref string requestString, string argument, string value)
+        {
+            var token = requestString.Contains("?") ? "&" : "?";
+            requestString = $"{requestString}{token}{argument}={WebUtility.UrlEncode(value)}";
```

Comment:
I don't disagree, but note this method is actually the same as it was before -- I just changed the `if/else` block into a one-liner with the `?` ternary operator, but the logic is exactly the same.

Basically, yes, there's more cleanup to do here, I'd just prefer to do it in a followup PR since this one is already fairly complex.